### ISSUE TITLE
NPE in TableDependencyTracker.getDependentTables()

### ISSUE THREAD

**Issue by saturnism** [GoogleCloudPlatform/google-cloud-spanner-hibernate#179]
Issue present in the master. Issue not present in released version 0.1.0.

From the Quarkus sample:
```
PROJECT_ID=wise-coyote-827 INSTANCE_ID=sample DATABASE_NAME=quarkus-sample java -jar target/quarkus-jpa-sample-1.1.0-SNAPSHOT-runner.jar
2019-12-17 10:16:49,167 WARNING [com.goo.aut.oau.DefaultCredentialsProvider] (Agroal_8065725461) Your application has authenticated using end user credentials from Google Cloud SDK. We recommend that most server applications use service accounts instead. If your application continues to use end user credentials from Cloud SDK, you might receive a "quota exceeded" or "API not enabled" error. For more information about service accounts, see https://cloud.google.com/docs/authentication/.
javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:113)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:67)
	at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:54)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:109)
	at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:57)
	at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder.startAllPersistenceUnits(HibernateOrmRecorder.java:82)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits23.deploy_0(HibernateOrmProcessor$startPersistenceUnits23.zig:70)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits23.deploy(HibernateOrmProcessor$startPersistenceUnits23.zig:36)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:134)
	at io.quarkus.runtime.Application.start(Application.java:94)
	at io.quarkus.runtime.Application.run(Application.java:218)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.NullPointerException
	at com.google.cloud.spanner.hibernate.schema.TableDependencyTracker.getDependentTables(TableDependencyTracker.java:81)
	at com.google.cloud.spanner.hibernate.SpannerTableExporter.buildSqlStrings(SpannerTableExporter.java:76)
	at com.google.cloud.spanner.hibernate.SpannerTableExporter.getSqlCreateStrings(SpannerTableExporter.java:59)
	at com.google.cloud.spanner.hibernate.SpannerTableExporter.getSqlCreateStrings(SpannerTableExporter.java:40)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.createTable(AbstractSchemaMigrator.java:279)
	at org.hibernate.tool.schema.internal.GroupedSchemaMigratorImpl.performTablesMigration(GroupedSchemaMigratorImpl.java:71)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.performMigration(AbstractSchemaMigrator.java:207)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.doMigration(AbstractSchemaMigrator.java:114)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:184)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:73)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:320)
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:65)
	... 12 more
2019-12-17 10:16:53,235 WARNING [com.goo.clo.spa.jdb.SpannerPool] (main) There is no Spanner registered for ConnectionOptions cloudspanner:/projects/wise-coyote-827/instances/sample/databases/quarkus-sample
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:206)
	at io.quarkus.runtime.Application.start(Application.java:94)
	at io.quarkus.runtime.Application.run(Application.java:218)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: javax.persistence.PersistenceException: [PersistenceUnit: default] Unable to build Hibernate SessionFactory
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:113)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:67)
	at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:54)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:109)
	at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:57)
	at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder.startAllPersistenceUnits(HibernateOrmRecorder.java:82)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits23.deploy_0(HibernateOrmProcessor$startPersistenceUnits23.zig:70)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits23.deploy(HibernateOrmProcessor$startPersistenceUnits23.zig:36)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:134)
	... 3 more
Caused by: java.lang.NullPointerException
	at com.google.cloud.spanner.hibernate.schema.TableDependencyTracker.getDependentTables(TableDependencyTracker.java:81)
	at com.google.cloud.spanner.hibernate.SpannerTableExporter.buildSqlStrings(SpannerTableExporter.java:76)
	at com.google.cloud.spanner.hibernate.SpannerTableExporter.getSqlCreateStrings(SpannerTableExporter.java:59)
	at com.google.cloud.spanner.hibernate.SpannerTableExporter.getSqlCreateStrings(SpannerTableExporter.java:40)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.createTable(AbstractSchemaMigrator.java:279)
	at org.hibernate.tool.schema.internal.GroupedSchemaMigratorImpl.performTablesMigration(GroupedSchemaMigratorImpl.java:71)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.performMigration(AbstractSchemaMigrator.java:207)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.doMigration(AbstractSchemaMigrator.java:114)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:184)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:73)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:320)
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:65)
	... 12 more
```

**Comment by meltsufin** [comment]
@dzou Can you take a look? We should really make all of these samples automatically tested by CI.

**Comment by dzou** [comment]
I'll take a look.

**Comment by dzou** [comment]
Hi guys, I've run this code through the debugger and the root issue appears to be that Quarkus is not utilizing our `SpannerSchemaManagementTool`; it doesn't seem to respect the property `hibernate.schema_management_tool` being set by our [SpannerServiceContributor](https://github.com/GoogleCloudPlatform/google-cloud-spanner-hibernate/blob/d742835ab780df7a5c576cb6ac12aee76d9670a2/google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerServiceContributor.java).

I'll send an email to the quarkus folks to ask about this.

**Comment by dzou** [comment]
Fixed by #183.
